### PR TITLE
Haiku Patches

### PR DESCRIPTION
**Pull request by bitigchi**
This PR is an attempt to update the [previous PR](https://github.com/vim/vim/pull/1856) to the latest Vim version. This version builds fine on bare install with `./configure [args] && make install`, but there are still a few quirks:

* GUI ~~does not~~ fails to build
* ~~`vimrc` & plugins do not work.~~
* ~~FreeBSD build test fails with dependency operator error~~

~~For plugins and `vimrc`, previous [haikuports recipe](https://github.com/haikuports/haikuports/blob/master/app-editors/vim/vim-8.1.1240.recipe) uses a workaround on line 150, but I was not able to successfully apply that to this version yet. I'm not sure how the code in `misc1.c` applies to this.~~

For the GUI, the previous version on Haikuports build fine, but I am still not sure what is wrong with the updated bare version. ~~I've made a couple of additions to the code to make sure Haiku-specific files are counted, alas no luck~~Currently the build fails due to configure-related issues. Any guidance on this would be appreciated!

To build on Haiku, just follow the instructions in `runtime/doc/os_haiku.txt`.
To try Haiku itself, feel free to download and install the latest nightly images from [here](https://download.haiku-os.org)!

I hope to improve the documentation once above two issues have been resolved.
Thank you!

### CODE REVIEW

**Comment by brammool** [comment]
BSD make doesn't like it: make[1]: "/tmp/cirrus-ci-build/src/Makefile" line 1395: Missing dependency operator
I guess you can't use +=

**Comment by bitigchi** [comment]
> BSD make doesn't like it: make[1]: "/tmp/cirrus-ci-build/src/Makefile" line 1395: Missing dependency operator
> I guess you can't use +=

I guess my fix did not work. :\

In the meantime, plugins and vimrc files work now, only major thing left is the GUI.

File: runtime/doc/eval.txt [hunk 1]
```diff
@@ -10640,7 +10640,7 @@ xor({expr}, {expr})					*xor()*
 <
 
 							*feature-list*
-There are three types of features:
+There are four types of features:
```
Comment (k-takata) [inline runtime/doc/eval.txt]: This change should be reverted.

Comment (bitigchi) [reply runtime/doc/eval.txt]: Fixed.

**Comment by bitigchi** [comment]
What is the proper way of auto-generating `configure` in Vim context? I was able to start building with GUI (failing though, output below) after an `autoconf`, but it overwrites the file in `src/`, instead of `src/auto`. Also the diff is quite different than the current file, it's not just the additions. See diff attached.

[configure_diff.txt](https://github.com/vim/vim/files/4180345/configure_diff.txt)

```sh
gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_HAIKU    -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1        -o objects/beval.o beval.c
In file included from structs.h:110,
                from vim.h:1847,
                from beval.c:11:
beval.c: In function 'get_beval_info':
beval.c:154:21: error: 'BalloonEval' {aka 'struct BalloonEvalStruct'} has no member named 'y'
 row = Y_2_ROW(beval->y);
                    ^~
gui.h:117:24: note: in definition of macro 'Y_2_ROW'
# define Y_2_ROW(y) (((y) - gui.border_offset) / gui.char_height)
                       ^
beval.c:155:21: error: 'BalloonEval' {aka 'struct BalloonEvalStruct'} has no member named 'x'
 col = X_2_COL(beval->x);
                    ^~
gui.h:113:24: note: in definition of macro 'X_2_COL'
# define X_2_COL(x) (((x) - gui.border_offset) / gui.char_width)
                       ^
Makefile:3218: recipe for target 'objects/beval.o' failed
make: *** [objects/beval.o] Error 1
```

**Comment by k-takata** [comment]
> What is the proper way of auto-generating `configure` in Vim context?

Try `make autoconf`.

**Comment by bitigchi** [comment]
> > What is the proper way of auto-generating `configure` in Vim context?
> 
> Try `make autoconf`.

This worked thanks, but the diff is the same though. It changes more than it (probably) should. GUI build fails with the same error above, CLI build completes fine.

**Comment by k-takata** [comment]
> It changes more than it (probably) should.

It may depend on version of autoconf.

**Comment by bitigchi** [comment]
@brammool, thanks for including. I will open an issue to track the GUI build failure issue.

**Pull request closed without merging**